templates/common/errs/unknownvar.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package errs contains errors that are shared across packages. It's named
+// this way to avoid colliding with "errors" (stdlib), "error" (a builtin type),
+// and "err" (a common variable name).
+package errs
+
+import "fmt"
+
+// UnknownVarError is an error that will be returned when a template
+// references a variable that's nonexistent.
+type UnknownVarError struct {
+	VarName       string
+	AvailableKeys []string
+	Wrapped       error
+}
```

Comment:
Nit: Do these all need to be public? It's a tad odd to have `Wrapped` and `Unwrap`.

Reply:
Since these errors are created in a few packages, we have two options:

 - Leave `Wrapped` public so it can be initialized
 - Make `Wrapped` private and provide a public constructor that initializes it.

I'm leaning towards the former just for simplicity.